packages/flutter/test/widgets/autocomplete_test.dart

```diff
@@ -647,6 +647,52 @@ void main() {
     );
   });
 
+  testWidgets('rebuilds the options while they are visible', (WidgetTester tester) async {
+    final GlobalKey fieldKey = GlobalKey();
+    final GlobalKey optionsKey = GlobalKey();
+    final List<String> availableOptions = List<String>.of(kOptions);
+    Iterable<String>? lastOptions;
+    Widget builder() {
+      return MaterialApp(
+        home: Scaffold(
+          body: RawAutocomplete<String>(
+            optionsBuilder: (TextEditingValue textEditingValue) {
+              lastOptions = List<String>.of(availableOptions).where((String option) {
+                return option.contains(textEditingValue.text.toLowerCase());
+              });
+              return lastOptions!;
+            },
+            optionsViewBuilder: (BuildContext context, AutocompleteOnSelected<String> onSelected, Iterable<String> options) {
+              return Container(key: optionsKey);
+            },
+            fieldViewBuilder: (BuildContext context, TextEditingController textEditingController, FocusNode focusNode, VoidCallback onFieldSubmitted) {
+              return TextField(
+                key: fieldKey,
+                autofocus: true,
+                focusNode: focusNode,
+                controller: textEditingController,
+              );
+            },
+          ),
+        ),
+      );
+    }
+
+    await tester.pumpWidget(builder());
+
+    // Enter text to show the options.
+    await tester.enterText(find.byKey(fieldKey), 'go');
+    await tester.pumpAndSettle();
+    expect(find.byKey(optionsKey), findsOneWidget);
+    expect(lastOptions, <String>['dingo', 'flamingo', 'goose']);
+
+    // Rebuild the options while they are visible.
+    availableOptions.add('gorilla');
+    await tester.pumpWidget(builder());
```

Comment:
I checked and this test does fail on master, but I'm worried that it doesn't exactly test how people will use this asynchronously.